HashTable.java

```diff
@@ -0,0 +1,107 @@
+class HashTable{
+
+    //I am using String array as a practice to see what happens when Strings are stored in HashMap/HashTable
+    String[] hashArr;
+    int size;
+    int arrSize;
+
+    public HashTable(int numOfSlots){
+        //This i read in some documentation that it is always better to have prime number as slots.
+        //The main reason can be due to the use of mod operator to calculate the index, prime number helps.
+
+        //So I am going to check if the number of slots are prime. if not then i will check the next available
+        //prime number and assign it to the the slot.
+        if (isPrime(numOfSlots)) {
+            hashArr = new String[numOfSlots];
+            arrSize = numOfSlots;
+        }
+        else{
+            int nextPrimeNumber = getNextPrimeNumber(numOfSlots);
+            hashArr = new String[nextPrimeNumber];
+            arrSize = nextPrimeNumber;
+        }
+    }
+
+    private boolean isPrime(int numOfSlots){
+
+        for (int i = 2; i*i <= numOfSlots; i++){
+            if (numOfSlots % i == 0){
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private int getNextPrimeNumber(int numOfSlots){
+        int i = numOfSlots;
+        while (true){
```

Comment:
A better way to use boolean logic would be **"while(! isPrime( i ))"**

Reply:
okay..that makes sense.